src/org/ods/component/ScanWithSnykStage.groovy

```diff
@@ -0,0 +1,88 @@
+package org.ods.component
+
+import org.ods.services.SnykService
+
+class ScanWithSnykStage extends Stage {
+  protected String STAGE_NAME = 'Snyk Security Scan'
+  protected SnykService snyk
+
+  ScanWithSnykStage(def script, IContext context, Map config, SnykService snyk) {
```

Comment:
I see an issue with this constructor implementation:
Map `config` is modified in the constructor. The caller of the constructor would have a map with different state after calling the constructor. This could be misleading side effect.

I think a better approach could be to not modify the map. The values of `failOnVulnerabilities`, `organisation`, `projectName`, `buildFile` could be stored as class fields with `private` scope.

Reply:
You are right. If we change it, we should clone the map though I think. In reality this isn't really a concern as the way you call this is through e.g. `odsComponentStageScanWithSnyk(context, [foo: bar])`, so you don't care about the modification.

Reply:
It might be that today is not a concern because the stage is called as you described. But the usage could change later. A developer that uses the stage will not expect the map to be changed. 
But actually the map is not needed internally. One could just copy from the map the values needed and store them in internal class variable to avoid any potential mistake coding. This is what I mean: 
```
  private String failOnVulnerabilities
  private String organisation
  private String projectName
  private String buildFile

  ScanWithSnykStage(def script, IContext context, Map config, SnykService snyk) {
    super(script, context, config)
    if (!config.containsKey('failOnVulnerabilities')) {
      failOnVulnerabilities = context.failOnSnykScanVulnerabilities
    }
    if (!config.organisation) {
      organisation = context.projectId
    }
    if (!config.projectName) {
      projectName = context.componentId
    }
    if (!config.buildFile) {
      buildFile = 'build.gradle'
    }
    this.snyk = snyk
  }
```